src/common/Thread.h

```diff
@@ -68,13 +70,14 @@ std::string get_thread_name(const std::thread& t);
 void kill(std::thread& t, int signal);
 
 template<typename Fun, typename... Args>
-std::thread make_named_thread(const std::string& s,
+std::thread make_named_thread(std::string_view n,
 			      Fun&& fun,
 			      Args&& ...args) {
-  auto t = std::thread(std::forward<Fun>(fun),
-		       std::forward<Args>(args)...);
-  set_thread_name(t, s);
-  return t;
-}
 
+  return std::thread([n = std::string(n)](Fun&& fun, Args&& ...args) mutable {
```

Comment:
This lambda doesn't need to be mutable anymore, does it?

Reply:
It does not, thank you.

Reply:
I think the types of the parameters are wrong too.  Since the arguments are perfectly forwarded, the parameters should be auto&&, otherwise it's bound to break at some point, like when passing a non-const lvalue instead of a temporary for Fun or various combinations of arguments for Args.

Reply:
Do you think so? I would have thought the Args&& expanded in the lambda should have the exact same types, reference, const, and all, as the Args&& in the original function signature.

Reply:
Looks like you were right. I'll change it.

Reply:
I highlighted that just because it looked a little weird to me, your reply made me look deeper.

It looks like that they do get expanded to the exact same types, but this is exactly the problem.  The arguments to the thread function appear to be copied/moved twice: first into a std::thread field, and then into a thread function.  Everything, including referenceness, gets stripped (see decay_copy() paragraph at https://en.cppreference.com/w/cpp/thread/thread/thread).  The thread function is called with rvalues, so fixing the type to something that can be a non-const lvalue reference doesn't work here...